### PR TITLE
[wasm] Emscripten 2.0.34 bump

### PR DESCRIPTION
**Pull request by radekdoulik**
Performance wise it looks similar to 2.0.23. Didn't notice any regression here. The managed startup is slightly slower, in second run I measured 223ms, so it is probably just measurement error.

| measurement | 2.0.23 time | 2.0.34 |
|-:|-:|-:|
|                    AppStart, Page show |    63.0000ms |    63.4458ms |
|                AppStart, Reach managed |   222.5600ms |   226.1304ms |
|        Exceptions, NoExceptionHandling |     0.0921us |     0.0843us |
|                   Exceptions, TryCatch |     0.0955us |     0.0885us |
|              Exceptions, TryCatchThrow |     0.0026ms |     0.0026ms |
|             Exceptions, TryCatchFilter |     0.1018us |     0.0929us |
|       Exceptions, TryCatchFilterInline |     0.0869us |     0.0800us |
|        Exceptions, TryCatchFilterThrow |     0.0033ms |     0.0034ms |
| Exceptions, TryCatchFilterThrowApplies |     0.0025ms |     0.0025ms |
|         Json, non-ASCII text serialize |     9.0376ms |     7.9638ms |
|       Json, non-ASCII text deserialize |    14.0856ms |    11.7460ms |
|                  Json, small serialize |     0.2351ms |     0.2203ms |
|                Json, small deserialize |     0.3556ms |     0.3359ms |
|                  Json, large serialize |    69.6667ms |    63.6420ms |
|                Json, large deserialize |    99.6792ms |    92.1754ms |
|              WebSocket, PartialSend 1B |     0.0017ms |     0.0017ms |
|            WebSocket, PartialSend 64KB |     0.0653ms |     0.0658ms |
|             WebSocket, PartialSend 1MB |     0.9636ms |     0.9364ms |
|           WebSocket, PartialReceive 1B |     0.0026ms |     0.0025ms |
|         WebSocket, PartialReceive 10KB |     0.0060ms |     0.0040ms |
|        WebSocket, PartialReceive 100KB |     0.0000us |     0.0000us |

### CODE REVIEW

**Comment by vargaz** [comment]
Might need this as well:
      <_EmccCommonFlags Include="-s ENVIRONMENT=&quot;web,webview,worker,node,shell&quot;" />

Newer versions of emscripten no longer build with 'shell' support by default, i.e. running under v8.

**Review by vargaz — APPROVED** [review]

**Comment by vargaz** [comment]
Fix for the build failure:
https://gist.github.com/vargaz/4bfce768ed19d283ddc31be124cc82a9

**Comment by radekdoulik** [comment]
Related: https://github.com/dotnet/runtime/issues/62708

File: src/mono/wasm/wasm.proj [hunk 1]
```diff
@@ -64,21 +66,23 @@
...
-      <_EmccCommonFlags Include="-s NO_EXIT_RUNTIME=1" />
-      <_EmccCommonFlags Include="-s FORCE_FILESYSTEM=1" />
-      <_EmccCommonFlags Include="-s EXPORTED_RUNTIME_METHODS=&quot;['FS','print','ccall','cwrap','setValue','getValue','UTF8ToString','UTF8ArrayToString','FS_createPath','FS_createDataFile','removeRunDependency','addRunDependency']&quot;" />
-      <_EmccCommonFlags Include="-s EXPORTED_FUNCTIONS=&quot;['_free','_malloc']&quot;" />
-      <_EmccCommonFlags Include="--source-map-base http://example.com" />
-
-      <_EmccCommonFlags Include="-s STRICT_JS=1" />      
-      <_EmccCommonFlags Include="-s EXPORT_NAME=&quot;'createDotnetRuntime'&quot;" />
-      <_EmccCommonFlags Include="-s MODULARIZE=1"/>
       <_EmccCommonFlags Include="-s EXPORT_ES6=1" Condition="'$(WasmEnableES6)' == 'true'" />
```
Comment (pavelsavara) [inline src/mono/wasm/wasm.proj:69]: EXPORT_ES6 is probably also link flag, right ?

**Comment by lewing** [comment]
looks like we need to bump Emscripten in icu https://github.com/dotnet/icu/pull/177

**Review by naricc — APPROVED** [review]

**Comment by SamMonoRT** [comment]
@naricc - any corresponding change needed anywhere in BDN or Perf repo ?

**Comment by lewing** [comment]
runtime (Libraries Build windows net48 x86 Release) failure is https://github.com/dotnet/runtime/issues/64153

File: src/mono/mono/mini/mini-generic-sharing.c [hunk 2]
```diff
@@ -1763,7 +1765,10 @@ mini_get_interp_in_wrapper (MonoMethodSignature *sig)
 	 * stack, pass this address to the interp_entry and when we return it we use
 	 * CEE_MONO_LDNATIVEOBJ
 	 */
-	return_native_struct = sig->ret->type == MONO_TYPE_VALUETYPE && sig->pinvoke && !sig->marshalling_disabled;
+#ifndef DISABLE_JIT
+	return_native_struct =
+#endif
```
Comment (BrzVlad) [inline src/mono/mono/mini/mini-generic-sharing.c:1770]: why endif here and not whole statement ?

**Review by lewing — APPROVED** [review]

**Comment by radical** [comment]
Failures:

- `Microsoft.Extensions.Logging.Generators.Roslyn3.11.Tests` - timing out, likely OOM
- https://github.com/dotnet/runtime/issues/64724
- https://github.com/dotnet/runtime/issues/64725
- https://github.com/dotnet/runtime/issues/64727

These will be investigated after this PR is merged.

**Comment by sbomer** [comment]
Why was this merged with known runtime-extra-platforms failures? I ask because we are trying to keep get runtime and runtime-extra-platforms to green to make it easier to detect new failures. Would it be reasonable to disable the failing tests before merging in the future? Just curious what we can do to meet our build health goals without unnecessarily slowing down development.

**Comment by radical** [comment]
> Why was this merged with known runtime-extra-platforms failures?

This was part of a bigger mess, which started with an update dependencies PR from emsdk getting merged, which bumped the emscripten version being used, which broken Wasm.Build.Tests (on `main`, and PRs), which got mixed with many helix queues getting removed, and all that combining to cause a badly red `main`, and lots of PRs being blocked. Merging this unblocked them, and made `main` much less red. And I opened a PR to disable those tests today (https://github.com/dotnet/runtime/pull/64759).

> I ask because we are trying to keep get runtime and runtime-extra-platforms to green to make it easier to detect new failures. Would it be reasonable to disable the failing tests before merging in the future? Just curious what we can do to meet our build health goals without unnecessarily slowing down development.

I completely agree with that, and avoid merging with any failures caused by the PR. But this was an unusual case.

**Comment by sbomer** [comment]
Thanks @radical, I appreciate the explanation

**Pull request merged**